### ISSUE TITLE
MatAutocompleteList Filtering improvements

### ISSUE THREAD

**Issue by sandrohanea** [SamProf/MatBlazor#547]
Add the possibility to handle large lists (e.g. using some database, using some API which filters data based on some text).

Proposed solution:
Having the next interface in MatBlazor:
```
public interface IAutocompleteSource<TItem>()
{
    IAsyncEnumerable<ITem> GetFilteredList(string inputText);
}
```

have a property of type IAutocompleteSource on the MatAutocompleteList which can be provided as a parameter instead of ItemsSource.
When the text value is changed, the items in the dialog will be updated based on the result of this alternative source if it exists.

What do you think about this?

If it's ok with you, I can implement this as I'm already using this approach in a Blazor Project.

**Comment by enkodellc** [comment]
@sandrohanea will this be another PR on top of your recent edits?

**Comment by sandrohanea** [comment]
Yes, this is an idea of an improvenent. But after thinking about it maybe it's better to be done on the new Autocomplete component which is currently in progress. Who is working on that? @SamProf ? Is there anything I can do to help?
